src/ape_ethereum/ecosystem.py

```diff
@@ -108,25 +82,19 @@ class StaticFeeTransaction(BaseTransaction):
     Transactions that are pre-EIP-1559 and use the ``gasPrice`` field.
     """
 
-    gas_price: int = None  # type: ignore
-    type: TransactionType = TransactionType.STATIC
+    gas_price: Optional[int] = Field(None, alias="gasPrice")
+    max_priority_fee: Optional[int] = Field(None, exclude=True)
+    type: TransactionType = Field(TransactionType.STATIC, exclude=True)
 
-    @property
-    def max_fee(self) -> int:
-        return (self.gas_limit or 0) * (self.gas_price or 0)
-
-    @max_fee.setter
-    def max_fee(self, valie):
-        raise NotImplementedError("Max fee is not settable for static-fee transactions.")
-
-    def as_dict(self):
-        data = super().as_dict()
-        if "gas_price" in data:
-            data["gasPrice"] = data.pop("gas_price")
-
-        data.pop("type")
+    def __init__(self, **data) -> None:
+        """
```

Comment:
doc strs in `__init__` compete the class-level doc strs.
You probably just want this as a normal comment.